### ISSUE TITLE
To deselect or to select not.

### ISSUE THREAD

**Issue by esjaytee** [Beep6581/RawTherapee#2931]
It's an issue or not: Food for thought. I work with a certain brand of cameras these days and my main workflow involves instant switch to Coiour tab after opening to image and setting manual WB just to see what's in it. (The picture. I guess Canon users agree the AWB is usually off by a mile or so and setting custom in the camera WB is.. well, not an option).

Nothing wrong with that. But then I switch back to Exposure tab. Should the dropper tool for custom WB be automatically deselected when switching tabs? I think so. What do You think?

Another issue or not: Should the tools that can be switched on/off be automatically switched on, when I move some of their controls? I'd like that. What do You think?

System: W7 x64, RT 4.2.363

Samppa

p.s. I'm fairly reluctant to fix anything that ain't broke, but damn Your good. RawTherapee seems to get features faster than one can learn ;)

**Comment by sguyader** [comment]
If it's easy to code this beahviour (regarding automatic deselection of the
WB eye dropper), I think it would be good. It happens quite often that when
I want to use the eyedropper for custom WB, I forget to deselect it and
then I want to navigate inside the image but when I click inside the
image... RT adjusts the WB to where I clicked. Not a big deal, but
sometimes annoying.

**Comment by raceviper13** [comment]
this has always been slightly inconvenient for me.  i would approve this change.

**Comment by Beep6581** [comment]
Before we continue, do you (plural) know deselecting the white balance picker is as simple as right-clicking anywhere on the preview?

**Comment by raceviper13** [comment]
no, i did not!  I will try to use that instead.  seems like it could be a low priority now.

**Comment by sguyader** [comment]
I didn't know either, but the "issue" is not that it is complicated to deselect, it's rather that after find the correct spot WB, we proceed to somthing else... and often we forget to that the eye dropper is still active and easily accidentally click on another spot of the image, which changes the previously selected WB (ctrl+z allows to go back to the last WB). So it's not a big issue.

**Comment by esjaytee** [comment]
Beep6581: Yes. But I seem to forget it a lot?! I'd say it comes from using several image/photo manipulation ect. packages (with different toolbar/dropdown based UI's) where selecting another tool cancels the previous.  I'd even say, it's a sort of a de facto standard behavior.

**Comment by Hombre57** [comment]
I'd rather change the cursor shape when the WB's pipette is active, not only displaying the pipette square. Having the same cross icon leads to this confusion IMHO. I prefer this solution over automatically deselecting, because we can't be sure that the user want to exit the function only because he switch to another tab. Yes, we could guess it, but some will probably complain about why it set it off in "this" case but not in "that" case.

**Comment by esjaytee** [comment]
Hombre57: Yes. You must be right. User can do a whole lotta things with WB dropper still selected. Even zoom in and out, but one thing user can't do, grab the picture to scroll it ;) With Navigator panel pointing to different parts of the picture is possible of course. Just my workflow WB->Exposure->1:1 Zoom to verify focus. If WB is still mile of after exposure compensation, I switch instantly back to Colour anyway ;)

What about: low priority: checkbox to the Preferences-> General -> Layout

Samppa

**Comment by Beep6581** [comment]
Checkbox for what?

**Comment by esjaytee** [comment]
Beep6581: For automatic deselection of WB dropper tool on tab change.

**Comment by Beep6581** [comment]
No, we won't be adding a checkbox for that. Petty stuff, bloat.

**Comment by esjaytee** [comment]
Sorry, You have a whole lotta checkboxes that do nothing in my GUI. What if You told me, what was the problem with Your implementation of the GUI? You can't catch a a click?

Samppa

**Comment by Hombre57** [comment]
There are clicks everywhere, and we have then to find out if it's one that deselect that button. Do you have an idea of how to do that sort of thing ? I know 2 or 3 things about GTK, but don't pretend to know everything about it, of course. Usually, we listen to some specific objects to get clic events (AFAIK, they do not propagate when they are processed by the widget you've clicked on), but what you ask is listening to all objects. I don't know how to do that.

**Comment by esjaytee** [comment]
Sorry, I'm no coder with only very limited gui patching experience.

But just to piss everybody off while trying to be witty I must say deselection works EXCELLENT when using "Select Straight Line" especially for landscape photography.

**Comment by mrx23dot** [comment]
Isn't the **most common** use case that we only use picker once after pressing Picker button?
Then manually fine tuning slider.
(at least this is how I do it)

Instead of randomly choosing many points after each other to find the perfect tone.
